src/components/NcDatetime/NcDatetime.vue

```diff
@@ -0,0 +1,221 @@
+ -
+ - @author Ferdinand Thiessen <opensource@fthiessen.de>
+ -
+ - @license AGPL-3.0-or-later
+ -
+ - This program is free software: you can redistribute it and/or modify
+ - it under the terms of the GNU Affero General Public License as
+ - published by the Free Software Foundation, either version 3 of the
+ - License, or (at your option) any later version.
+ -
+ - This program is distributed in the hope that it will be useful,
+ - but WITHOUT ANY WARRANTY; without even the implied warranty of
+ - MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ - GNU Affero General Public License for more details.
+ -
+ - You should have received a copy of the GNU Affero General Public License
+ - along with this program. If not, see <http://www.gnu.org/licenses/>.
+ -
+ -->
+
+<docs>
+
+### General description
+
+This components purpose is to display a timestamp in the users local time format.
+It also supports relative time, for examples *6 seconds ago*.
+
+#### Standard usage
+
+Without any optional parameters the timestamp is displayed as a relative datetime and a title with the full date is added.
+
+```vue
+<template>
+	<NcDatetime :timestamp="timestamp" />
+</template>
+<script>
+	export default {
+		data() {
+			return {
+				timestamp: Date.now(),
+			}
+		},
+	}
+</script>
+```
+
+#### Custom date or time format
+
+The component allows to format the full date for the title by settings the `format` property.
+It is also possible to disable relative time by setting the `relativeTime` property to `false`.
+
+```vue
+<template>
+	<div>
+		<h4>Short text version</h4>
+		<NcDatetime :timestamp="timestamp" relative-time="short" />
+
+		<h4>Custom title format</h4>
+		<NcDatetime :timestamp="timestamp" :format="timeFormat" />
+
+		<h4>Without relative time</h4>
+		<NcDatetime :timestamp="timestamp" :format="timeFormat" :relative-time="false" />
+	</div>
+</template>
+<script>
+	export default {
+		data() {
+			return {
+				timestamp: Date.now(),
+				/** For allowed formats see the Intl.DateTimeFormat options */
+				timeFormat: {
+					dateStyle: 'short',
+					timeStyle: 'full'
+				},
+			}
+		},
+	}
+</script>
+<style>
+h4 {
+	font-weight: bold;
+	margin-top: 12px;
+}
+</style>
+```
+</docs>
+
+<template>
+	<span class="nc-datetime"
+		:data-timestamp="timestamp"
+		:title="formattedFullTime">{{ formattedTime }}</span>
+</template>
+
+<script>
+import { getCanonicalLocale } from '@nextcloud/l10n'
+
+export default {
+	name: 'NcDatetime',
+
+	props: {
+		/**
+		 * The timestamp to display, either an unix timestamp (in milliseconds) or a Date object
+		 */
+		timestamp: {
+			type: [Date, Number],
+			required: true,
+		},
+		/**
+		 * The format used for displaying, or if relative time is used the format used for the title (optional)
+		 *
+		 * @type {Intl.DateTimeFormatOptions}
+		 */
+		format: {
+			type: Object,
+			default: () => ({ timeStyle: 'medium', dateStyle: 'short' }),
```

Comment:
Wouldn't it make sense to separate these into two separate props?

Reply:
Which two properties? title format and main format?
Or do you mean `timeStyle` and `dateStyle`?

If the latter one, then I do not think so. As there are more options possible I just allowed every `Intl.DateTimeFormatOptions` option (which allow finer control over the style by setting the format of `weekday, hour, month...` directly).